ERCS/erc-7760.md

```diff
@@ -0,0 +1,511 @@
+---
+eip: 7760
+title: Minimal Upgradeable Proxies
+description: Minimal upgradeable proxies with immutable arguments and support for onchain implementation queries
+author: Atarpara (@Atarpara), JT Riley (@jtriley-eth), xiaobaiskill (@xiaobaiskill), Vectorized (@Vectorized)
+discussions-to: https://ethereum-magicians.org/t/erc-7760-minimal-upgradeable-proxies/20868
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-08-19
+requires: 1967
+---
+
+## Abstract
+
+This standard defines minimal [ERC-1967](./eip-1967.md) proxies for three patterns: (1) transparent, (2) UUPS, (3) beacon. The proxies support optional immutable arguments which are appended to the end of their runtime bytecode. Additional variants which support onchain implementation querying are provided.
+
+## Motivation
+
+Having standardized minimal bytecode for upgradeable proxies enables the following:
+
+1. Automatic verification on block explorers.
+2. Ability for immutable arguments to be queried onchain, as these arguments are stored at the same bytecode offset,
+3. Ability for the implementation to be queried and verified onchain.
+
+The minimal nature of the proxies enables cheaper deployment and runtime costs.
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Overview
+
+All of the following proxies MAY have optional data bytecode appended to the end of their runtime bytecode. 
+
+Emitting the ERC-1967 events during initialization is OPTIONAL. Indexers MUST NOT expect the initialization code to emit the ERC-1967 events.
+
+### Minimal ERC-1967 transparent upgradeable proxy
+
+The transparent upgradeable proxy MUST be deployed by a factory that is responsible for authenticating upgrades.
+
+As the proxy's runtime bytecode contains logic to allow the factory to set any storage slot with any value, the initialization code MAY skip storing the implementation slot.
+
+The upgrading logic does not emit the ERC-1967 event. Indexers MUST NOT expect the upgrading logic to emit the ERC-1967 events.
+
+During upgrades, the factory MUST call the upgradeable proxy with following calldata:
+
+```solidity
+abi.encodePacked(
+    // The new implementation address, converted to a 32-byte word.
+    uint256(uint160(implementation)),
+    // ERC-1967 implementation slot.
+    bytes32(0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc),
```

Comment:
Can you share how was 0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc calculated, assuming its a named storage slot (https://eips.ethereum.org/EIPS/eip-7201) or is it generated by random?